### PR TITLE
refactor: remove usage of `<Box>` and `sx`

### PR DESCRIPTION
**Pull request by aslilac**
Part of #10612

- Use the `FC` type much more consistently throughout the codebase for declaring components and their prop types

- Remove all usage of the `sx` prop throughout the codebase

- Remove all usage of `Box` throughout the codebase

- Break lots of inline type definitions out into their own declarations

- Make some small `css` styles inline and break some bigger ones out into `const` declarations

### CODE REVIEW

**Comment by Parkreiner** [comment]
Looks really good! Nothing worth blocking over, but I did flag what are likely a few styling bugs/typos

**Review by buenos-nachos — APPROVED** [review]
Looks really good! Nothing worth blocking over, but I did flag what are likely a few styling bugs/typos

File: site/src/pages/TemplateSettingsPage/TemplateSchedulePage/ScheduleDialog.tsx [hunk 1]
```diff
@@ -71,9 +71,7 @@ export const ScheduleDialog: FC<PropsWithChildren<ScheduleDialogProps>> = ({
                 <div css={styles.dialogDescription}>{`
                 This change will result in ${inactiveWorkspacesToGoDormant} workspaces being immediately transitioned to the dormant state and ${inactiveWorkspacesToGoDormantInWeek} over the next seven days. To prevent this, do you want to reset the inactivity period for all template workspaces?`}</div>
                 <FormControlLabel
-                  sx={{
-                    marginTop: 2,
-                  }}
+                  css={{ marginTop: 2 }}
```
[the remark this answers is not in the document]
Comment (aslilac) [reply site/src/pages/TemplateSettingsPage/TemplateSchedulePage/ScheduleDialog.tsx:74]: yes!

File: site/src/pages/TerminalPage/TerminalPage.tsx [hunk 2]
```diff
@@ -358,22 +361,22 @@ const BottomBar = ({ proxy, latency }: { proxy: Region; latency?: number }) => {
...
-            <Box
-              sx={{
+            <div
+              css={{
                 height: 6,
                 width: 6,
                 backgroundColor: color,
                 border: 0,
-                borderRadius: 9999,
+                borderRadius: 3,
```
[the remark this answers is not in the document]
Comment (aslilac) [reply site/src/pages/TerminalPage/TerminalPage.tsx:370]: it's not actually a visual change, I just like to be precise :^) 3 is half of 6, so it's so still "fully rounded"

**Pull request merged**